mapping/mapping_vectors.go

```diff
@@ -37,45 +37,97 @@ func NewVectorFieldMapping() *FieldMapping {
 	}
 }
 
-func (fm *FieldMapping) processVector(propertyMightBeVector interface{},
-	pathString string, path []string, indexes []uint64, context *walkContext) {
-	propertyVal := reflect.ValueOf(propertyMightBeVector)
-	if !propertyVal.IsValid() {
-		return
+// validate and process a flat vector
+func processFlatVector(vecI interface{}, dims int) ([]float32, bool) {
+	vecV := reflect.ValueOf(vecI)
+	if !vecV.IsValid() || vecV.Kind() != reflect.Slice || vecV.Len() != dims {
```

Comment:
If we mandate caller of processFlatVector to make sure that vecI is valid, is slice and len equal dims.
Then, we can avoid checking it here.

Reply:
done